pkg/resourcesets/collector.go

```diff
@@ -248,157 +174,147 @@ func (h *ResourceHandler) filterByLabel(ctx context.Context, dr dynamic.Resource
+			continue
+		}
+		if resourceNameRegexpStr != "" {
+			nameMatched := resourceNameRegexpStr == "." || resourceNameRegexp.MatchString(resourceName)
+			if !nameMatched {
+				logrus.Debugf("Skipping [%s] because it did not match ResourceNameRegexp [%s]", resourceName, filter.ResourceNameRegexp)
+			} else if excludeResourceNameRegexp != nil && excludeResourceNameRegexp.MatchString(resourceName) {
+				logrus.Debugf("Skipping [%s] because it matched both ResourceNameRegexp and ExcludeResourceNameRegexp [%s]", resourceName, filter.ExcludeResourceNameRegexp)
+			} else {
+				filteredByName = append(filteredByName, resObj)
 			}
-		} else {
-			filteredByName = filteredByResourceNames
 		}
-
 	}
+
 	return filteredByName, nil
 }
 
 func (h *ResourceHandler) filterByNamespace(filter v1.ResourceSelector, filteredByName []unstructured.Unstructured) ([]unstructured.Unstructured, error) {
-	var filteredByNamespace, filteredByNamespaceRegex, filteredObjects []unstructured.Unstructured
-	filteredByNsMap := make(map[*unstructured.Unstructured]bool)
+	var filteredObjects []unstructured.Unstructured
+	var namespaceRegexp *regexp.Regexp
+	var excludeNamespaceRegexp *regexp.Regexp
 
-	if len(filter.Namespaces) > 0 {
-		logrus.Debugf("Using Namespaces %s to filter namespaces", strings.Join(filter.Namespaces, ","))
-		allowedNamespaces := make(map[string]bool)
-		for _, ns := range filter.Namespaces {
-			allowedNamespaces[ns] = true
-		}
-		for _, resObj := range filteredByName {
-			metadata := resObj.Object["metadata"].(map[string]interface{})
-			ns := metadata["namespace"].(string)
-			if allowedNamespaces[ns] {
-				filteredByNamespace = append(filteredByNamespace, resObj)
-				filteredByNsMap[&resObj] = true
-			}
-		}
+	if len(filter.Namespaces) == 0 && filter.NamespaceRegexp == "" {
+		return filteredByName, nil
 	}
 	if filter.NamespaceRegexp != "" {
+		var err error
 		logrus.Debugf("Using NamespaceRegexp %s to filter resource names", filter.NamespaceRegexp)
-		if filter.NamespaceRegexp == "." {
-			// "." will match all namespaces, so return all objects obtained after filtering by name
-			return filteredByName, nil
-		}
-		for _, resObj := range filteredByName {
-			metadata := resObj.Object["metadata"].(map[string]interface{})
-			ns := metadata["namespace"].(string)
-			nsMatched, err := regexp.MatchString(filter.NamespaceRegexp, ns)
-			if err != nil {
-				return filteredByNamespace, err
-			}
-			if !nsMatched {
-				continue
-			}
-			filteredByNamespaceRegex = append(filteredByNamespaceRegex, resObj)
+		namespaceRegexp, err = regexp.Compile(filter.NamespaceRegexp)
+		if err != nil {
+			return filteredObjects, err
 		}
-
-		if len(filteredByNamespaceRegex) == 0 {
-			// none matched regex
-			// return whatever was filtered by exact namespace match
-			// if that list is also empty, it means no namespaces matched the given filters
-			return filteredByNamespace, nil
+	}
+	if filter.ExcludeResourceNameRegexp != "" {
```

Comment:
This section of code is within `filterByNamespace` so we shouldn't be touching `ExcludeResourceNameRegexp` anymore. Since that's only necessary in the `filterByName` method. Think we need 292-299 removed.

Reply:
Done